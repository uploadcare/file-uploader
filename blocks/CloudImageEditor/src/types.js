/**
 * Mapping of loading resources per operation
 *
 * @typedef {Map<String, Map<String, boolean>>} LoadingOperations
 */

/**
 * Image size
 *
 * @typedef {{ width: Number; height: Number }} ImageSize
 */

/**
 * @typedef {Object} Rectangle
 * @property {Number} x
 * @property {Number} y
 * @property {Number} width
 * @property {Number} height
 */

/**
 * @typedef {{
 *   enhance?: number;
 *   brightness?: number;
 *   exposure?: number;
 *   gamma?: number;
 *   contrast?: number;
 *   saturation?: number;
 *   vibrance?: number;
 *   warmth?: number;
 *   rotate?: number;
 *   mirror?: boolean;
 *   flip?: boolean;
 *   filter?: { name: string; amount: number };
 *   crop?: { dimensions: [number, number]; coords: [number, number] };
 * }} Transformations
 */

/**
 * @typedef {Object} ApplyResult
 * @property {string} originalUrl
 * @property {string} cdnUrlModifiers
 * @property {string} cdnUrl
 * @property {Transformations} transformations
 */

/**
 * @typedef {Object} ChangeResult
 * @property {string} originalUrl
 * @property {string} cdnUrlModifiers
 * @property {string} cdnUrl
 * @property {Transformations} transformations
 */

/** @typedef {{ type: 'aspect-ratio'; width: number; height: number }} CropAspectRatio */

/** @typedef {CropAspectRatio[]} CropPresetList */

/**
 * @typedef {Partial<{
 *   [K in Direction]: {
 *     direction: Direction;
 *     pathNode: SVGElement;
 *     interactionNode: SVGElement;
 *     groupNode: SVGElement;
 *   };
 * }>} FrameThumbs
 */

/** @typedef {'' | 'n' | 's' | 'e' | 'w' | 'ne' | 'nw' | 'se' | 'sw'} Direction */

export {};
