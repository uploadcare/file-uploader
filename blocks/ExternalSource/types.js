// @ts-check

/**
 * @typedef {{
 *   error?: undefined;
 *   alternatives?: Record<string, string>;
 *   is_image?: boolean | null;
 *   filename?: string;
 *   obj_type: 'selected_file';
 *   url: string;
 * }} DoneSuccessResponse
 */

/**
 * @typedef {{
 *   'selected-files-change': {
 *     type: 'selected-files-change';
 *     total: number;
 *     selectedCount: number;
 *   } & (
 *     | {
 *         isReady: false;
 *         isMultipleMode: boolean;
 *         selectedFiles: undefined;
 *       }
 *     | {
 *         isReady: true;
 *         isMultipleMode: true;
 *         selectedFiles: DoneSuccessResponse[];
 *       }
 *     | {
 *         isReady: true;
 *         isMultipleMode: false;
 *         selectedFiles: [DoneSuccessResponse] | [];
 *       }
 *   );
 *   'toolbar-state-change': {
 *     type: 'toolbar-state-change';
 *     isVisible: boolean;
 *   };
 * }} InputMessageMap
 */

/** @typedef {keyof InputMessageMap} InputMessageType */
/** @typedef {InputMessageMap[InputMessageType]} InputMessage */

/**
 * @template {import('./types').InputMessageType} T
 * @typedef {(message: import('./types').InputMessageMap[T]) => void} InputMessageHandler
 */

/**
 * @typedef {{
 *   '--uc-font-family': string;
 *   '--uc-font-size': string;
 *   '--uc-line-height': string;
 *   '--uc-button-size': string;
 *   '--uc-preview-size': string;
 *   '--uc-input-size': string;
 *   '--uc-padding': string;
 *   '--uc-radius': string;
 *   '--uc-transition': string;
 *   '--uc-background': string;
 *   '--uc-foreground': string;
 *   '--uc-primary': string;
 *   '--uc-primary-hover': string;
 *   '--uc-primary-transparent': string;
 *   '--uc-primary-foreground': string;
 *   '--uc-secondary': string;
 *   '--uc-secondary-hover': string;
 *   '--uc-secondary-foreground': string;
 *   '--uc-muted': string;
 *   '--uc-muted-foreground': string;
 *   '--uc-destructive': string;
 *   '--uc-destructive-foreground': string;
 *   '--uc-border': string;
 *   '--uc-primary-rgb-light': string;
 *   '--uc-primary-light': string;
 *   '--uc-primary-hover-light': string;
 *   '--uc-primary-transparent-light': string;
 *   '--uc-background-light': string;
 *   '--uc-foreground-light': string;
 *   '--uc-primary-foreground-light': string;
 *   '--uc-secondary-light': string;
 *   '--uc-secondary-hover-light': string;
 *   '--uc-secondary-foreground-light': string;
 *   '--uc-muted-light': string;
 *   '--uc-muted-foreground-light': string;
 *   '--uc-destructive-light': string;
 *   '--uc-destructive-foreground-light': string;
 *   '--uc-border-light': string;
 *   '--uc-primary-rgb-dark': string;
 *   '--uc-primary-dark': string;
 *   '--uc-primary-hover-dark': string;
 *   '--uc-primary-transparent-dark': string;
 *   '--uc-background-dark': string;
 *   '--uc-foreground-dark': string;
 *   '--uc-primary-foreground-dark': string;
 *   '--uc-secondary-dark': string;
 *   '--uc-secondary-hover-dark': string;
 *   '--uc-secondary-foreground-dark': string;
 *   '--uc-muted-dark': string;
 *   '--uc-muted-foreground-dark': string;
 *   '--uc-destructive-dark': string;
 *   '--uc-destructive-foreground-dark': string;
 *   '--uc-border-dark': string;
 *   '--uc-primary-oklch-light': string;
 *   '--uc-primary-oklch-dark': string;
 * }} ThemeDefinition
 */

/**
 * @typedef {{
 *       type: 'select-all';
 *     }
 *   | {
 *       type: 'deselect-all';
 *     }
 *   | {
 *       type: 'set-theme-definition';
 *       theme: Record<string, string>;
 *     }
 *   | {
 *       type: 'set-locale-definition';
 *       localeDefinition: string;
 *     }
 *   | {
 *       type: 'set-embed-css';
 *       css: string;
 *     }} OutputMessage
 */

export {};
