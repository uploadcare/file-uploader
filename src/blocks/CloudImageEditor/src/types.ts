/**
 * Mapping of loading resources per operation
 */
export type LoadingOperations = Map<string, Map<string, boolean>>;

/**
 * Image size
 */
export interface ImageSize {
  width: number;
  height: number;
}

export interface Rectangle {
  x: number;
  y: number;
  width: number;
  height: number;
}

export interface Transformations {
  enhance?: number;
  brightness?: number;
  exposure?: number;
  gamma?: number;
  contrast?: number;
  saturation?: number;
  vibrance?: number;
  warmth?: number;
  rotate?: number;
  mirror?: boolean;
  flip?: boolean;
  filter?: { name: string; amount: number };
  crop?: { dimensions: [number, number]; coords: [number, number] };
}

export interface ApplyResult {
  originalUrl: string;
  cdnUrlModifiers: string;
  cdnUrl: string;
  transformations: Transformations;
}

export type ChangeResult = ApplyResult;

export interface CropAspectRatio {
  type: 'aspect-ratio';
  width: number;
  height: number;
  id: string;
  hasFreeform?: boolean;
}

export type CropPresetList = CropAspectRatio[];

export type Direction = '' | 'n' | 's' | 'e' | 'w' | 'ne' | 'nw' | 'se' | 'sw';

export type FrameThumbs = Partial<
  Record<
    Direction,
    {
      direction: Direction;
      pathNode: SVGElement;
      interactionNode: SVGElement;
      groupNode: SVGElement;
    }
  >
>;
