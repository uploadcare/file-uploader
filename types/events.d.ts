import type { EventPayload } from '../blocks/UploadCtxProvider/EventEmitter';

export type EventMap = {
  [T in keyof EventPayload]: CustomEvent<EventPayload[T]>;
};
