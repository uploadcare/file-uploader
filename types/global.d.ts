import { UC_WINDOW_KEY } from '../abstract/connectBlocksFrom.js';
import * as blocks from '../index.js';

declare global {
  interface Window {
    [UC_WINDOW_KEY]?: typeof blocks;
  }
}
