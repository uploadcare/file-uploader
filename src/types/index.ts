export type * from './events';
export type * from './exported';
