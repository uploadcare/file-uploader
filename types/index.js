export * from './exported.js';
export * from './events.js';
