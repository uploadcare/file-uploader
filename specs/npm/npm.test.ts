import { describe, expect, test } from 'vitest';

const stickyPackageVersion = (obj: any) => {
  if ('PACKAGE_VERSION' in obj) {
    return {
      ...obj,
      PACKAGE_VERSION: '0.0.0-stub',
    };
  }
  return obj;
};

describe('NPM package', () => {
  test('import asserts are working', async () => {
    await expect(import(`@uploadcare/file-uploader/${'abstract/Block.js'}`)).rejects.toThrow();
  });

  test('root ssr stubs export should match snapshot', async () => {
    const rootModule = await import('@uploadcare/file-uploader');
    expect('IS_SSR_STUBS' in rootModule).toBe(true);
    expect(stickyPackageVersion(rootModule)).toMatchSnapshot('root export');
  });

  test('export `abstract/loadFileUploaderFrom.js` should match snapshot', async () => {
    const loadFileUploaderFrom = await import('@uploadcare/file-uploader/abstract/loadFileUploaderFrom.js');
    expect(stickyPackageVersion(loadFileUploaderFrom)).toMatchSnapshot('abstract/loadFileUploaderFrom.js');
  });

  test('export `env` should match snapshot', async () => {
    const env = await import('@uploadcare/file-uploader/env');
    expect(stickyPackageVersion(env)).toMatchSnapshot('env');
  });

  test('exports `web` should match snapshot', async () => {
    const webBundles = [
      'file-uploader.iife.min.js',
      'file-uploader.min.js',
      'uc-basic.min.css',
      'uc-cloud-image-editor.min.css',
      'uc-cloud-image-editor.min.js',
      'uc-file-uploader-inline.min.css',
      'uc-file-uploader-inline.min.js',
      'uc-file-uploader-minimal.min.css',
      'uc-file-uploader-minimal.min.js',
      'uc-file-uploader-regular.min.css',
      'uc-file-uploader-regular.min.js',
      'uc-img.min.js',
    ];

    for (const bundle of webBundles) {
      const m = await import(`@uploadcare/file-uploader/web/${bundle}`);
      expect(stickyPackageVersion(m)).toMatchSnapshot(`web/${bundle}`);
    }
  });

  test('jsx types should be exported', async () => {
    await expect(
      // @ts-expect-error
      import('@uploadcare/file-uploader/types/jsx'),
    ).resolves.not.toThrow();
  });

  test('locales should be exported', async () => {
    await expect(import('@uploadcare/file-uploader/locales/file-uploader/en.js')).resolves.not.toThrow();
  });
});
