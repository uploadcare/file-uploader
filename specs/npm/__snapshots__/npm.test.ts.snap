// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`NPM package > export \`abstract/loadFileUploaderFrom.js\` should match snapshot > abstract/loadFileUploaderFrom.js 1`] = `
{
  "UC_WINDOW_KEY": "UC",
  "loadFileUploaderFrom": [Function],
}
`;

exports[`NPM package > export \`env\` should match snapshot > env 1`] = `
{
  "PACKAGE_NAME": "blocks",
  "PACKAGE_VERSION": "0.0.0-stub",
}
`;

exports[`NPM package > exports \`web\` should match snapshot > web/file-uploader.iife.min.js 1`] = `{}`;

exports[`NPM package > exports \`web\` should match snapshot > web/file-uploader.min.js 1`] = `
{
  "ActivityBlock": [Function],
  "ActivityHeader": [Function],
  "BaseComponent": [Function],
  "Block": [Function],
  "BtnUi": [Function],
  "CameraSource": [Function],
  "CloudImageEditor": [Function],
  "CloudImageEditorActivity": [Function],
  "CloudImageEditorBlock": [Function],
  "Config": [Function],
  "Copyright": [Function],
  "CropFrame": [Function],
  "Data": [Function],
  "DropArea": [Function],
  "EditorAspectRatioButtonControl": [Function],
  "EditorCropButtonControl": [Function],
  "EditorFilterControl": [Function],
  "EditorFreeformButtonControl": [Function],
  "EditorImageCropper": [Function],
  "EditorImageFader": [Function],
  "EditorOperationControl": [Function],
  "EditorScroller": [Function],
  "EditorSlider": [Function],
  "EditorToolbar": [Function],
  "ExternalSource": [Function],
  "ExternalUploadSource": {
    "BOX": "box",
    "DROPBOX": "dropbox",
    "EVERNOTE": "evernote",
    "FACEBOOK": "facebook",
    "FLICKR": "flickr",
    "GDRIVE": "gdrive",
    "GPHOTOS": "gphotos",
    "HUDDLE": "huddle",
    "ONEDRIVE": "onedrive",
    "VK": "vk",
  },
  "FileItem": [Function],
  "FileUploaderInline": [Function],
  "FileUploaderMinimal": [Function],
  "FileUploaderRegular": [Function],
  "FormInput": [Function],
  "Icon": [Function],
  "Img": [Function],
  "LineLoaderUi": [Function],
  "Modal": [Function],
  "ModalEvents": {
    "ADD": "modal:add",
    "CLOSE": "modal:close",
    "CLOSE_ALL": "modal:closeAll",
    "DELETE": "modal:delete",
    "DESTROY": "modal:destroy",
    "OPEN": "modal:open",
  },
  "PACKAGE_NAME": "blocks",
  "PACKAGE_VERSION": "0.0.0-stub",
  "PresenceToggle": [Function],
  "ProgressBar": [Function],
  "ProgressBarCommon": [Function],
  "Select": [Function],
  "SimpleBtn": [Function],
  "SliderUi": [Function],
  "SolutionBlock": [Function],
  "SourceBtn": [Function],
  "SourceList": [Function],
  "Spinner": [Function],
  "StartFrom": [Function],
  "Thumb": [Function],
  "UID": [Function],
  "UploadCtxProvider": [Function],
  "UploadList": [Function],
  "UploadSource": {
    "API": "js-api",
    "BOX": "box",
    "CAMERA": "camera",
    "DRAW": "draw",
    "DROPBOX": "dropbox",
    "DROP_AREA": "drop-area",
    "EVERNOTE": "evernote",
    "EXTERNAL": "external",
    "FACEBOOK": "facebook",
    "FLICKR": "flickr",
    "GDRIVE": "gdrive",
    "GPHOTOS": "gphotos",
    "HUDDLE": "huddle",
    "LOCAL": "local",
    "MOBILE_PHOTO_CAMERA": "mobile-photo-camera",
    "MOBILE_VIDEO_CAMERA": "mobile-video-camera",
    "ONEDRIVE": "onedrive",
    "URL": "url",
    "VK": "vk",
  },
  "UploaderBlock": [Function],
  "UrlSource": [Function],
  "defineComponents": [Function],
  "defineLocale": [Function],
  "loadFileUploaderFrom": [Function],
  "toKebabCase": [Function],
}
`;

exports[`NPM package > exports \`web\` should match snapshot > web/uc-basic.min.css 1`] = `
{
  "default": "",
}
`;

exports[`NPM package > exports \`web\` should match snapshot > web/uc-cloud-image-editor.min.css 1`] = `
{
  "default": "",
}
`;

exports[`NPM package > exports \`web\` should match snapshot > web/uc-cloud-image-editor.min.js 1`] = `
{
  "BtnUi": [Function],
  "CloudImageEditor": [Function],
  "CloudImageEditorBlock": [Function],
  "Config": [Function],
  "CropFrame": [Function],
  "EditorAspectRatioButtonControl": [Function],
  "EditorCropButtonControl": [Function],
  "EditorFilterControl": [Function],
  "EditorFreeformButtonControl": [Function],
  "EditorImageCropper": [Function],
  "EditorImageFader": [Function],
  "EditorOperationControl": [Function],
  "EditorScroller": [Function],
  "EditorSlider": [Function],
  "EditorToolbar": [Function],
  "Icon": [Function],
  "LineLoaderUi": [Function],
  "PresenceToggle": [Function],
  "SliderUi": [Function],
  "defineComponents": [Function],
}
`;

exports[`NPM package > exports \`web\` should match snapshot > web/uc-file-uploader-inline.min.css 1`] = `
{
  "default": "",
}
`;

exports[`NPM package > exports \`web\` should match snapshot > web/uc-file-uploader-inline.min.js 1`] = `
{
  "ActivityBlock": [Function],
  "ActivityHeader": [Function],
  "BaseComponent": [Function],
  "Block": [Function],
  "BtnUi": [Function],
  "CameraSource": [Function],
  "CloudImageEditor": [Function],
  "CloudImageEditorActivity": [Function],
  "CloudImageEditorBlock": [Function],
  "Config": [Function],
  "Copyright": [Function],
  "CropFrame": [Function],
  "Data": [Function],
  "DropArea": [Function],
  "EditorAspectRatioButtonControl": [Function],
  "EditorCropButtonControl": [Function],
  "EditorFilterControl": [Function],
  "EditorFreeformButtonControl": [Function],
  "EditorImageCropper": [Function],
  "EditorImageFader": [Function],
  "EditorOperationControl": [Function],
  "EditorScroller": [Function],
  "EditorSlider": [Function],
  "EditorToolbar": [Function],
  "ExternalSource": [Function],
  "ExternalUploadSource": {
    "BOX": "box",
    "DROPBOX": "dropbox",
    "EVERNOTE": "evernote",
    "FACEBOOK": "facebook",
    "FLICKR": "flickr",
    "GDRIVE": "gdrive",
    "GPHOTOS": "gphotos",
    "HUDDLE": "huddle",
    "ONEDRIVE": "onedrive",
    "VK": "vk",
  },
  "FileItem": [Function],
  "FileUploaderInline": [Function],
  "FileUploaderMinimal": [Function],
  "FileUploaderRegular": [Function],
  "FormInput": [Function],
  "Icon": [Function],
  "Img": [Function],
  "LineLoaderUi": [Function],
  "Modal": [Function],
  "ModalEvents": {
    "ADD": "modal:add",
    "CLOSE": "modal:close",
    "CLOSE_ALL": "modal:closeAll",
    "DELETE": "modal:delete",
    "DESTROY": "modal:destroy",
    "OPEN": "modal:open",
  },
  "PACKAGE_NAME": "blocks",
  "PACKAGE_VERSION": "0.0.0-stub",
  "PresenceToggle": [Function],
  "ProgressBar": [Function],
  "ProgressBarCommon": [Function],
  "Select": [Function],
  "SimpleBtn": [Function],
  "SliderUi": [Function],
  "SolutionBlock": [Function],
  "SourceBtn": [Function],
  "SourceList": [Function],
  "Spinner": [Function],
  "StartFrom": [Function],
  "Thumb": [Function],
  "UID": [Function],
  "UploadCtxProvider": [Function],
  "UploadList": [Function],
  "UploadSource": {
    "API": "js-api",
    "BOX": "box",
    "CAMERA": "camera",
    "DRAW": "draw",
    "DROPBOX": "dropbox",
    "DROP_AREA": "drop-area",
    "EVERNOTE": "evernote",
    "EXTERNAL": "external",
    "FACEBOOK": "facebook",
    "FLICKR": "flickr",
    "GDRIVE": "gdrive",
    "GPHOTOS": "gphotos",
    "HUDDLE": "huddle",
    "LOCAL": "local",
    "MOBILE_PHOTO_CAMERA": "mobile-photo-camera",
    "MOBILE_VIDEO_CAMERA": "mobile-video-camera",
    "ONEDRIVE": "onedrive",
    "URL": "url",
    "VK": "vk",
  },
  "UploaderBlock": [Function],
  "UrlSource": [Function],
  "defineComponents": [Function],
  "defineLocale": [Function],
  "loadFileUploaderFrom": [Function],
  "toKebabCase": [Function],
}
`;

exports[`NPM package > exports \`web\` should match snapshot > web/uc-file-uploader-minimal.min.css 1`] = `
{
  "default": "",
}
`;

exports[`NPM package > exports \`web\` should match snapshot > web/uc-file-uploader-minimal.min.js 1`] = `
{
  "ActivityBlock": [Function],
  "ActivityHeader": [Function],
  "BaseComponent": [Function],
  "Block": [Function],
  "BtnUi": [Function],
  "CameraSource": [Function],
  "CloudImageEditor": [Function],
  "CloudImageEditorActivity": [Function],
  "CloudImageEditorBlock": [Function],
  "Config": [Function],
  "Copyright": [Function],
  "CropFrame": [Function],
  "Data": [Function],
  "DropArea": [Function],
  "EditorAspectRatioButtonControl": [Function],
  "EditorCropButtonControl": [Function],
  "EditorFilterControl": [Function],
  "EditorFreeformButtonControl": [Function],
  "EditorImageCropper": [Function],
  "EditorImageFader": [Function],
  "EditorOperationControl": [Function],
  "EditorScroller": [Function],
  "EditorSlider": [Function],
  "EditorToolbar": [Function],
  "ExternalSource": [Function],
  "ExternalUploadSource": {
    "BOX": "box",
    "DROPBOX": "dropbox",
    "EVERNOTE": "evernote",
    "FACEBOOK": "facebook",
    "FLICKR": "flickr",
    "GDRIVE": "gdrive",
    "GPHOTOS": "gphotos",
    "HUDDLE": "huddle",
    "ONEDRIVE": "onedrive",
    "VK": "vk",
  },
  "FileItem": [Function],
  "FileUploaderInline": [Function],
  "FileUploaderMinimal": [Function],
  "FileUploaderRegular": [Function],
  "FormInput": [Function],
  "Icon": [Function],
  "Img": [Function],
  "LineLoaderUi": [Function],
  "Modal": [Function],
  "ModalEvents": {
    "ADD": "modal:add",
    "CLOSE": "modal:close",
    "CLOSE_ALL": "modal:closeAll",
    "DELETE": "modal:delete",
    "DESTROY": "modal:destroy",
    "OPEN": "modal:open",
  },
  "PACKAGE_NAME": "blocks",
  "PACKAGE_VERSION": "0.0.0-stub",
  "PresenceToggle": [Function],
  "ProgressBar": [Function],
  "ProgressBarCommon": [Function],
  "Select": [Function],
  "SimpleBtn": [Function],
  "SliderUi": [Function],
  "SolutionBlock": [Function],
  "SourceBtn": [Function],
  "SourceList": [Function],
  "Spinner": [Function],
  "StartFrom": [Function],
  "Thumb": [Function],
  "UID": [Function],
  "UploadCtxProvider": [Function],
  "UploadList": [Function],
  "UploadSource": {
    "API": "js-api",
    "BOX": "box",
    "CAMERA": "camera",
    "DRAW": "draw",
    "DROPBOX": "dropbox",
    "DROP_AREA": "drop-area",
    "EVERNOTE": "evernote",
    "EXTERNAL": "external",
    "FACEBOOK": "facebook",
    "FLICKR": "flickr",
    "GDRIVE": "gdrive",
    "GPHOTOS": "gphotos",
    "HUDDLE": "huddle",
    "LOCAL": "local",
    "MOBILE_PHOTO_CAMERA": "mobile-photo-camera",
    "MOBILE_VIDEO_CAMERA": "mobile-video-camera",
    "ONEDRIVE": "onedrive",
    "URL": "url",
    "VK": "vk",
  },
  "UploaderBlock": [Function],
  "UrlSource": [Function],
  "defineComponents": [Function],
  "defineLocale": [Function],
  "loadFileUploaderFrom": [Function],
  "toKebabCase": [Function],
}
`;

exports[`NPM package > exports \`web\` should match snapshot > web/uc-file-uploader-regular.min.css 1`] = `
{
  "default": "",
}
`;

exports[`NPM package > exports \`web\` should match snapshot > web/uc-file-uploader-regular.min.js 1`] = `
{
  "ActivityBlock": [Function],
  "ActivityHeader": [Function],
  "BaseComponent": [Function],
  "Block": [Function],
  "BtnUi": [Function],
  "CameraSource": [Function],
  "CloudImageEditor": [Function],
  "CloudImageEditorActivity": [Function],
  "CloudImageEditorBlock": [Function],
  "Config": [Function],
  "Copyright": [Function],
  "CropFrame": [Function],
  "Data": [Function],
  "DropArea": [Function],
  "EditorAspectRatioButtonControl": [Function],
  "EditorCropButtonControl": [Function],
  "EditorFilterControl": [Function],
  "EditorFreeformButtonControl": [Function],
  "EditorImageCropper": [Function],
  "EditorImageFader": [Function],
  "EditorOperationControl": [Function],
  "EditorScroller": [Function],
  "EditorSlider": [Function],
  "EditorToolbar": [Function],
  "ExternalSource": [Function],
  "ExternalUploadSource": {
    "BOX": "box",
    "DROPBOX": "dropbox",
    "EVERNOTE": "evernote",
    "FACEBOOK": "facebook",
    "FLICKR": "flickr",
    "GDRIVE": "gdrive",
    "GPHOTOS": "gphotos",
    "HUDDLE": "huddle",
    "ONEDRIVE": "onedrive",
    "VK": "vk",
  },
  "FileItem": [Function],
  "FileUploaderInline": [Function],
  "FileUploaderMinimal": [Function],
  "FileUploaderRegular": [Function],
  "FormInput": [Function],
  "Icon": [Function],
  "Img": [Function],
  "LineLoaderUi": [Function],
  "Modal": [Function],
  "ModalEvents": {
    "ADD": "modal:add",
    "CLOSE": "modal:close",
    "CLOSE_ALL": "modal:closeAll",
    "DELETE": "modal:delete",
    "DESTROY": "modal:destroy",
    "OPEN": "modal:open",
  },
  "PACKAGE_NAME": "blocks",
  "PACKAGE_VERSION": "0.0.0-stub",
  "PresenceToggle": [Function],
  "ProgressBar": [Function],
  "ProgressBarCommon": [Function],
  "Select": [Function],
  "SimpleBtn": [Function],
  "SliderUi": [Function],
  "SolutionBlock": [Function],
  "SourceBtn": [Function],
  "SourceList": [Function],
  "Spinner": [Function],
  "StartFrom": [Function],
  "Thumb": [Function],
  "UID": [Function],
  "UploadCtxProvider": [Function],
  "UploadList": [Function],
  "UploadSource": {
    "API": "js-api",
    "BOX": "box",
    "CAMERA": "camera",
    "DRAW": "draw",
    "DROPBOX": "dropbox",
    "DROP_AREA": "drop-area",
    "EVERNOTE": "evernote",
    "EXTERNAL": "external",
    "FACEBOOK": "facebook",
    "FLICKR": "flickr",
    "GDRIVE": "gdrive",
    "GPHOTOS": "gphotos",
    "HUDDLE": "huddle",
    "LOCAL": "local",
    "MOBILE_PHOTO_CAMERA": "mobile-photo-camera",
    "MOBILE_VIDEO_CAMERA": "mobile-video-camera",
    "ONEDRIVE": "onedrive",
    "URL": "url",
    "VK": "vk",
  },
  "UploaderBlock": [Function],
  "UrlSource": [Function],
  "defineComponents": [Function],
  "defineLocale": [Function],
  "loadFileUploaderFrom": [Function],
  "toKebabCase": [Function],
}
`;

exports[`NPM package > exports \`web\` should match snapshot > web/uc-img.min.js 1`] = `
{
  "Img": [Function],
}
`;

exports[`NPM package > root ssr stubs export should match snapshot > root export 1`] = `
{
  "ActivityBlock": [Function],
  "ActivityHeader": [Function],
  "BaseComponent": [Function],
  "Block": [Function],
  "BtnUi": [Function],
  "CameraSource": [Function],
  "CloudImageEditor": [Function],
  "CloudImageEditorActivity": [Function],
  "CloudImageEditorBlock": [Function],
  "Config": [Function],
  "Copyright": [Function],
  "CropFrame": [Function],
  "Data": [Function],
  "DropArea": [Function],
  "EditorAspectRatioButtonControl": [Function],
  "EditorCropButtonControl": [Function],
  "EditorFilterControl": [Function],
  "EditorFreeformButtonControl": [Function],
  "EditorImageCropper": [Function],
  "EditorImageFader": [Function],
  "EditorOperationControl": [Function],
  "EditorScroller": [Function],
  "EditorSlider": [Function],
  "EditorToolbar": [Function],
  "ExternalSource": [Function],
  "ExternalUploadSource": {
    "BOX": "box",
    "DROPBOX": "dropbox",
    "EVERNOTE": "evernote",
    "FACEBOOK": "facebook",
    "FLICKR": "flickr",
    "GDRIVE": "gdrive",
    "GPHOTOS": "gphotos",
    "HUDDLE": "huddle",
    "ONEDRIVE": "onedrive",
    "VK": "vk",
  },
  "FileItem": [Function],
  "FileUploaderInline": [Function],
  "FileUploaderMinimal": [Function],
  "FileUploaderRegular": [Function],
  "FormInput": [Function],
  "IS_SSR_STUBS": true,
  "Icon": [Function],
  "Img": [Function],
  "LineLoaderUi": [Function],
  "Modal": [Function],
  "ModalEvents": {
    "ADD": "modal:add",
    "CLOSE": "modal:close",
    "CLOSE_ALL": "modal:closeAll",
    "DELETE": "modal:delete",
    "DESTROY": "modal:destroy",
    "OPEN": "modal:open",
  },
  "PACKAGE_NAME": "blocks",
  "PACKAGE_VERSION": "0.0.0-stub",
  "PresenceToggle": [Function],
  "ProgressBar": [Function],
  "ProgressBarCommon": [Function],
  "Select": [Function],
  "SimpleBtn": [Function],
  "SliderUi": [Function],
  "SolutionBlock": [Function],
  "SourceBtn": [Function],
  "SourceList": [Function],
  "Spinner": [Function],
  "StartFrom": [Function],
  "Thumb": [Function],
  "UID": [Function],
  "UploadCtxProvider": [Function],
  "UploadList": [Function],
  "UploadSource": {
    "API": "js-api",
    "BOX": "box",
    "CAMERA": "camera",
    "DRAW": "draw",
    "DROPBOX": "dropbox",
    "DROP_AREA": "drop-area",
    "EVERNOTE": "evernote",
    "EXTERNAL": "external",
    "FACEBOOK": "facebook",
    "FLICKR": "flickr",
    "GDRIVE": "gdrive",
    "GPHOTOS": "gphotos",
    "HUDDLE": "huddle",
    "LOCAL": "local",
    "MOBILE_PHOTO_CAMERA": "mobile-photo-camera",
    "MOBILE_VIDEO_CAMERA": "mobile-video-camera",
    "ONEDRIVE": "onedrive",
    "URL": "url",
    "VK": "vk",
  },
  "UploaderBlock": [Function],
  "UrlSource": [Function],
  "defineComponents": [Function],
  "defineLocale": [Function],
  "loadFileUploaderFrom": [Function],
  "toKebabCase": [Function],
}
`;
